radicle-cob/src/object/collaboration.rs

```diff
@@ -25,28 +25,80 @@ pub use list::list;
 mod update;
 pub use update::{update, Update};
 
+/// The full object identifier for a [`CollaborativeObject`].
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct ObjectIdentifier {
+    /// The [`TypeName`] for the given [`CollaborativeObject`].
+    pub name: TypeName,
+    /// The [`ObjectId`] for the given [`CollaborativeObject`].
+    pub object: ObjectId,
+}
```

Comment:
Hmm, I feel like ObjectIdentifier stands for ObjectId; ie. they are too close and confusable. Some potential suggestions:

* `TypedObjectId`
* `TypedId`
* `TObjectId`

Reply:
Or ObjectMetadata?  The TypeName is descriptive right?  Or wow does the type name affect the identity of the object?  This domain is new to me.

Reply:
It basically describes the type of object refered to by the object-id, so like `(Issue, 123)`

Reply:
What about `CobId`, since it's the combination of the two that identifies a `CollaborativeObject`?

Reply:
That could work, though there isn't as clear a relation betwen ObjectId and CobId :thinking:

Reply:
Interestingly though, I haven't found that I "missed" that type while porting some of the code so far, so I wonder if we should scrap the type idea, and just put the function in some module hierarchy that makes sense?

Reply:
@cloudhead I'm not quite understanding, could you make the idea more concrete?